### PR TITLE
Relay runtime: fix Store constructor parameters

### PR DESCRIPTION
**Pull request by mrtnzlml**
- [x] Use a meaningful title for the pull request. Include the name of the package modified.
- [ ] Test the change in your own code. (Compile and run.)
- [ ] Add or edit tests to reflect the change. (Run with `npm test`.)
- [x] Follow the advice from the [readme](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#make-a-pull-request).
- [x] Avoid [common mistakes](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#common-mistakes).
- [x] Run `npm run lint package-name` (or `tsc` if no `tslint.json` is present).

Select one of these and delete the others:

If changing an existing definition:
- [x] Provide a URL to documentation or source code which provides context for the suggested changes: https://github.com/facebook/relay/commit/6ed264413ba8cdd586d695e5ed234951ee9eca13
- [x] If this PR brings the type definitions up to date with a new version of the JS library, update the version number in the header.
- [x] If you are making substantial changes, consider adding a `tslint.json` containing `{ "extends": "dtslint/dt.json" }`. If for reason the any rule need to be disabled, disable it for that line using `// tslint:disable-next-line [ruleName]` and not for whole package so that the need for disabling can be reviewed.

### CODE REVIEW

**Comment by mrtnzlml** [comment]
NB: I am unable to run the tests locally on latest master (seems unrelated):

```text
➜  DefinitelyTyped git:(relay-runtime-fix-store-options) n test

> definitely-typed@0.0.3 test /Users/mrtnzlml/Work/DefinitelyTyped/DefinitelyTyped
> node node_modules/types-publisher/bin/tester/test.js --run-from-definitely-typed

Clean data
Clean logs
Clean output
Using local Definitely Typed at /Users/mrtnzlml/Work/DefinitelyTyped/DefinitelyTyped.
Parsing definitions...
Found 6441 packages.
Parsing in parallel...
Error: In package.json: Dependency connect-mongo not in whitelist.
If you are depending on another `@types` package, do *not* add it to a `package.json`. Path mapping should make the import work.
For namespaced dependencies you then have to add a `paths` mapping from `@namespace/library` to `namespace__library` in `tsconfig.json`.
If this is an external library that provides typings,  please make a pull request to types-publisher adding it to `dependenciesWhitelist.txt`.
    at checkPackageJsonDependencies (/Users/mrtnzlml/Work/DefinitelyTyped/DefinitelyTyped/node_modules/types-publisher/src/lib/definition-parser.ts:216:19)
    at combineDataForAllTypesVersions (/Users/mrtnzlml/Work/DefinitelyTyped/DefinitelyTyped/node_modules/types-publisher/src/lib/definition-parser.ts:98:37)
    at Object.getTypingInfo (/Users/mrtnzlml/Work/DefinitelyTyped/DefinitelyTyped/node_modules/types-publisher/src/lib/definition-parser.ts:24:24)
    at /Users/mrtnzlml/Work/DefinitelyTyped/DefinitelyTyped/node_modules/types-publisher/src/lib/definition-parser-worker.ts:18:30
    at Object.logUncaughtErrors (/Users/mrtnzlml/Work/DefinitelyTyped/DefinitelyTyped/node_modules/types-publisher/src/util/util.ts:99:38)
    at process.<anonymous> (/Users/mrtnzlml/Work/DefinitelyTyped/DefinitelyTyped/node_modules/types-publisher/src/lib/definition-parser-worker.ts:16:9)
    at process.emit (events.js:210:5)
    at process.emit (/Users/mrtnzlml/Work/DefinitelyTyped/DefinitelyTyped/node_modules/source-map-support/source-map-support.js:439:21)
    at emit (internal/child_process.js:876:12)
    at processTicksAndRejections (internal/process/task_queues.js:81:21)
Error: Parsing failed.
    at fail (/Users/mrtnzlml/Work/DefinitelyTyped/DefinitelyTyped/node_modules/types-publisher/src/util/util.ts:307:20)
    at ChildProcess.<anonymous> (/Users/mrtnzlml/Work/DefinitelyTyped/DefinitelyTyped/node_modules/types-publisher/src/util/util.ts:295:21)
    at ChildProcess.emit (events.js:210:5)
    at finish (internal/child_process.js:861:14)
    at processTicksAndRejections (internal/process/task_queues.js:75:11)
npm ERR! Test failed.  See above for more details.
```

**Comment by Stephen2** [comment]
I can't run `npm run test` on latest master either.  Not sure why, and am to novice to know what to do here.

File: types/relay-runtime/lib/store/RelayModernStore.d.ts [hunk 1]
```diff
@@ -22,7 +21,11 @@ import {
 } from './RelayConnection';
 
 export class RelayModernStore implements Store {
-    constructor(source: MutableRecordSource, gcScheduler?: Scheduler, operationLoader?: OperationLoader | null);
+    constructor(source: MutableRecordSource, options?: {
+        gcScheduler?: Scheduler | null,
```
Comment (Stephen2) [inline types/relay-runtime/lib/store/RelayModernStore.d.ts]: `gcScheduler?: ?Scheduler` in Flow means `gcScheduler?: Scheduler | null | undefined` more accurately, I think.

File: types/relay-runtime/lib/store/RelayModernStore.d.ts [hunk 2]
```diff
@@ -22,7 +21,11 @@ import {
...
+        operationLoader?: OperationLoader | null
+        gcReleaseBufferSize?: number | null,
```
Comment (Stephen2) [inline types/relay-runtime/lib/store/RelayModernStore.d.ts]: Same here, `| undefined`, if I'm right

File: types/relay-runtime/relay-runtime-tests.tsx [hunk 3]
```diff
@@ -16,6 +16,9 @@ import {
 
 const source = new RecordSource();
 const store = new Store(source);
+const storeWithOptions = new Store(source, {
+    gcReleaseBufferSize: 10,
+});
```
Comment (Stephen2) [inline types/relay-runtime/relay-runtime-tests.tsx]: I suppose the combinations here could be more thorough.  Not sure how far we generally want to go with test cases.  If we're going for completeness, we should do a bit more here.

**Review by Stephen2 — CHANGES_REQUESTED** [review]
Only `Request changes` to let the bot know what's going on.  My comments are more questions than blockers.

Question: is this constructor change released in `relay-runtime` yet?  Seems like it would be a breaking change and wouldn't release into relay8 ?

File: types/relay-runtime/lib/store/RelayModernStore.d.ts (hunk 2 above)
Comment (mrtnzlml) [reply types/relay-runtime/lib/store/RelayModernStore.d.ts]: 👍 you are right

Comment (mrtnzlml) [reply types/relay-runtime/lib/store/RelayModernStore.d.ts]: Ah, so actually no - TS behaves differently:

```text
ERROR: 25:42  no-redundant-undefined  Property is optional, so no need to specify `undefined` as a possible value. See: https://github.com/Microsoft/dtslint/blob/master/docs/no-redundant-undefined.md
```

Comment (mrtnzlml) [reply types/relay-runtime/lib/store/RelayModernStore.d.ts]: https://github.com/Microsoft/dtslint/blob/master/docs/no-redundant-undefined.md

File: types/relay-runtime/relay-runtime-tests.tsx (hunk 3 above)
Comment (mrtnzlml) [reply types/relay-runtime/relay-runtime-tests.tsx]: I added some more tests. There are many possible combinations though so I'd keep it simple unless there is some problem I didn't think about. :)

**Comment by mrtnzlml** [comment]
> Question: is this constructor change released in relay-runtime yet? Seems like it would be a breaking change and wouldn't release into relay8 ?

It's already in version 7.0.0, see: https://github.com/facebook/relay/commit/6ed264413ba8cdd586d695e5ed234951ee9eca13 and https://github.com/facebook/relay/releases/tag/v7.0.0

Relay TS definition files are quite behind in some cases. :)

**Comment by alloy** [comment]
I’ll wait on @Stephen2 for another pass, but otherwise this looks good to me 👍

**Comment by Stephen2** [comment]
> It's already in version 7.0.0, see: [facebook/relay@6ed2644](https://github.com/facebook/relay/commit/6ed264413ba8cdd586d695e5ed234951ee9eca13) and https://github.com/facebook/relay/releases/tag/v7.0.0
> 
> Relay TS definition files are quite behind in some cases. :)

:scream:

**Comment by Stephen2** [comment]
Thanks mate this is great :+1:

**Review by Stephen2 — APPROVED** [review]

**Pull request merged**